src/Service/MigrationService.php

```diff
@@ -0,0 +1,638 @@
+     * specify it explicitly like `--fields="id_key:primaryKey,name:string(12):notNull:unique"`
+     *
+     * @param array $value
+     */
+    public function fields(array $value): void
+    {
+        $this->fields = $value;
+    }
+
+    /**
+     * Set maximum length of a migration name.
+     *
+     * @param int $value
+     */
+    public function maxNameLength(int $value): void
+    {
+        $this->maxNameLength = $value;
+    }
+
+    /**
+     * List of namespaces containing the migration classes.
+     *
+     * @param array $value
+     *
+     * Migration namespaces should be resolvable as a [path alias](guide:concept-aliases) if prefixed with `@`, e.g.
+     * if you specify the namespace `app\migrations`, the code `$this->aliases->get('@app/migrations')` should be able
+     * to return the file path to the directory this namespace refers to.
+     *
+     * This corresponds with the [autoloading conventions](guide:concept-autoloading) of Yii.
+     */
+    public function migrationNamespaces(array $value): void
+    {
+        $this->migrationNamespaces = $value;
+    }
+
+    /**
+     * The directory containing the migration classes.
+     *
+     * This can be either a [path alias](guide:concept-aliases) or a directory path.
+     *
+     * Migration classes located at this path should be declared without a namespace.
+     * Use {@see migrationNamespaces} property in case you are using namespaced migrations.
+     *
+     * If you have set up {migrationNamespaces}, you may set this field to `null` in order to disable usage of
+     * migrations that are not namespaced.
+     *
+     * In general, to load migrations from different locations, {migrationNamespaces} is the preferable solution as the
+     * migration name contains the origin of the migration in the history, which is not the case when using multiple
+     * migration paths.
+     *
+     * @param array $value
+     *
+     * {@see $migrationNamespaces}
+     */
+    public function migrationPath(array $value): void
+    {
+        $this->migrationPath = $value;
+    }
+
+    /**
+     * Set the name of the table for keeping applied migration information.
+     *
+     * @param string $value the name of the table for keeping applied migration information.
+     */
+    public function migrationTable(string $value): void
+    {
+        $this->migrationTable = $value;
+    }
+
+    /**
+     * Set the value of templateFile
+     *
+     * @param string $value
+     */
+    public function templateFile(string $value): void
+    {
+        $this->templateFile = $value;
+    }
+
+    /**
+     * Indicates whether the table names generated should consider the `tablePrefix` setting of the DB connection.
+     *
+     * For example, if the table name is `post` the generator wil return `{{%post}}`.
+     *
+     * @param bool $value
+     */
+    public function useTablePrefix(bool $value): void
```

Comment:
Not used?

Reply:
Yes used config enabled/disabled `{{ table }}`